packages/open-api-gateway/src/construct/integrations/integrations.ts

```diff
@@ -0,0 +1,284 @@
+   * @see https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-swagger-extensions-integration-responses.html
+   */
+  readonly responses?: {
+    [responseStatusPattern: string]: ApiGatewayIntegrationResponse;
+  };
+  /**
+   * Custom timeout between 50 and 29,000 milliseconds. The default value is 29,000 milliseconds or 29 seconds.
+   */
+  readonly timeoutInMillis?: number;
+  /**
+   * Specifies the TLS configuration for an integration.
+   * @see https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-extensions-integration-tls-config.html
+   */
+  readonly tlsConfig?: ApiGatewayIntegrationTlsConfig;
+}
+
+/**
+ * Properties for rendering an integration into an API Gateway OpenAPI extension
+ */
+export interface IntegrationRenderProps {
+  /**
+   * The ID of the operation being rendered
+   */
+  readonly operationId: string;
+  /**
+   * The scope in which the integration is being rendered
+   */
+  readonly scope: Construct;
+}
+
+/**
+ * Properties for granting the API access to invoke the operation
+ */
+export interface IntegrationGrantProps {
+  /**
+   * The ID of the operation for which permissions are being granted
+   */
+  readonly operationId: string;
+  /**
+   * The scope in which permission resources can be created
+   */
+  readonly scope: Construct;
+  /**
+   * The api to grant permissions for
+   */
+  readonly api: SpecRestApi;
+}
+
+/**
+ * An integration for an API operation. You can extend this to implement your own integration if you like.
+ */
+export abstract class Integration {
+  /**
+   * Render the integration into an API Gateway OpenAPI extension
+   */
+  public abstract render(props: IntegrationRenderProps): ApiGatewayIntegration;
+  /**
+   * Grant permissions for the API to invoke the integration
+   */
+  public grant(_props: IntegrationGrantProps) {
+    // No permissions to grant by default
+  }
+}
+
+/**
+ * A lambda integration
+ */
+export class LambdaIntegration extends Integration {
+  private readonly lambdaFunction: IFunction;
+
+  constructor(lambdaFunction: IFunction) {
+    super();
+    this.lambdaFunction = lambdaFunction;
+  }
+
+  /**
+   * Render the lambda integration as a snippet of OpenAPI
+   */
+  public render(props: IntegrationRenderProps): ApiGatewayIntegration {
+    return {
+      type: "AWS_PROXY",
+      httpMethod: "POST",
+      uri: functionInvocationUri(props.scope, this.lambdaFunction),
+      passthroughBehavior: "WHEN_NO_MATCH",
+    };
+  }
+
+  /**
+   * Grant API Gateway permissions to invoke the lambda
+   */
+  public grant({ scope, api, operationId }: IntegrationGrantProps) {
+    new CfnPermission(scope, `LambdaPermission-${operationId}`, {
+      action: "lambda:InvokeFunction",
+      principal: "apigateway.amazonaws.com",
+      functionName: this.lambdaFunction.functionArn,
+      sourceArn: Stack.of(scope).formatArn({
+        service: "execute-api",
+        resource: api.restApiId,
+        resourceName: "*/*",
```

Comment:
given one integration is servicing a single route, shouldn't this be specific to said route?

Reply:
Yes it's likely possible to scope down - would require a bit of extra work to translate the method+path to a resource (eg we'd need to substitute path parameters with `*`) so might be best saved for a future enhancement? :)

Reply:
Updated to scope down to the specific stage/method/path :)